src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs

```diff
@@ -0,0 +1,143 @@
+
+            return new HeavyweightRenameLocations(
+               Symbol,
+               Solution,
+               Options,
+               FallbackOptions,
+               Locations,
+               implicitLocations,
+               referencedSymbols);
+        }
+
+        /// <summary>
+        /// Find the locations that need to be renamed.  Can cross process boundaries efficiently to do this.
+        /// </summary>
+        public static async Task<LightweightRenameLocations> FindRenameLocationsAsync(
+            ISymbol symbol, Solution solution, SymbolRenameOptions options, CodeCleanupOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfNull(solution);
+            Contract.ThrowIfNull(symbol);
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using (Logger.LogBlock(FunctionId.Renamer_FindRenameLocationsAsync, cancellationToken))
+            {
+                if (SerializableSymbolAndProjectId.TryCreate(symbol, solution, cancellationToken, out var serializedSymbol))
+                {
+                    var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
+                    if (client != null)
+                    {
+                        var result = await client.TryInvokeAsync<IRemoteRenamerService, SerializableRenameLocations?>(
+                            solution,
+                            (service, solutionInfo, callbackId, cancellationToken) => service.FindRenameLocationsAsync(solutionInfo, callbackId, serializedSymbol, options, cancellationToken),
+                            callbackTarget: new RemoteOptionsProvider<CodeCleanupOptions>(solution.Workspace.Services, fallbackOptions),
+                            cancellationToken).ConfigureAwait(false);
+
+                        if (result.HasValue && result.Value != null)
+                        {
+                            var rehydrated = await TryRehydrateAsync(
+                                solution, symbol, fallbackOptions, result.Value, cancellationToken).ConfigureAwait(false);
+
+                            if (rehydrated != null)
+                                return rehydrated;
+                        }
+
+                        // TODO: do not fall back to in-proc if client is available (https://github.com/dotnet/roslyn/issues/47557)
+                    }
+                }
+            }
+
+            // Couldn't effectively search in OOP. Perform the search in-proc.
+            var renameLocations = await HeavyweightRenameLocations.FindLocationsInCurrentProcessAsync(
+                symbol, solution, options, fallbackOptions, cancellationToken).ConfigureAwait(false);
+
+            return new LightweightRenameLocations(
+                symbol, solution, options, fallbackOptions, renameLocations.Locations,
```

Comment:
Should we dehydrate the symbol here?

Reply:
will do followup to make this clearer.

Reply:
followup here: https://github.com/dotnet/roslyn/pull/62803